app/grandchallenge/challenges/tasks.py

```diff
@@ -1,37 +1,55 @@
 from celery import shared_task
 from django.core.mail import mail_managers
+from django.db.models import Count, Max
 from requests import exceptions, get
 
 from grandchallenge.challenges.models import Challenge, ExternalChallenge
 from grandchallenge.evaluation.models import Evaluation
+from grandchallenge.profiles.admin import User
 from grandchallenge.subdomains.utils import reverse
 
 
 @shared_task
 def update_challenge_results_cache():
-    for c in Challenge.objects.all():
-        kwargs = {
-            "cached_num_participants": c.participants_group.user_set.all().count()
-        }
-
-        challenge_results = Evaluation.objects.filter(
-            submission__phase__challenge=c, published=True
-        ).order_by("-created")
+    challenges = Challenge.objects.all()
+    evaluation_info = (
+        Evaluation.objects.filter(rank__gt=0, published=True)
+        .values("submission__phase__challenge_id")
+        .annotate(
+            cached_num_results=Count("submission__phase__challenge_id"),
+            cached_latest_result=Max("created"),
+        )
+    )
+    evaluation_info_by_challenge = {
+        str(v["submission__phase__challenge_id"]): v for v in evaluation_info
+    }
+    participant_counts = User.objects.values(
+        "groups__participants_of_challenge"
+    ).annotate(cached_num_participants=Count("pk"))
```

Comment:
Just for my understanding, does this work? It doesn't need to be `"groups__participants_of_challenge_id"` and `Count("groups__participants_of_challenge_id")`?

Reply:
No, it doesn't. Actually in the other query, it would also suffice to call `submission__phase__challenge`. Is it clearer though to retrieve the id?

Reply:
The "Zen of Python" (`import this`) would tell us that "Explicit is better than implicit.", so I tend to err on the side of adding the id, but, if it works it works, I trust the Django developers to keep the API stable and if they do change the behaviour we would definitely hear about it before we hit issues.